frontend/packages/dev-console/src/components/health-checks/AddHealthCheckPage.tsx

```diff
@@ -0,0 +1,152 @@
+  }
+
+  const container = _.filter(
+    resource.data.spec.template.spec.containers,
+    (data) => data.name === currentContainer,
+  );
+
+  if (_.isEmpty(container)) {
+    return <div className="text-center">Container not found</div>;
+  }
+
+  const handleSubmit = (values, actions) => {
+    const {
+      healthChecks: { readinessProbe, livenessProbe, startupProbe },
+      containerName,
+    } = values;
+    const updatedResource = _.cloneDeep(resource.data);
+    const containerIndex = _.findIndex(updatedResource.spec.template.spec.containers, [
+      'name',
+      containerName,
+    ]);
+    updatedResource.spec.template.spec.containers[containerIndex] = {
+      ...container[containerIndex],
+      ...(readinessProbe.enabled
+        ? { readinessProbe: constructProbeData(readinessProbe.data) }
+        : {}),
+      ...(livenessProbe.enabled ? { livenessProbe: constructProbeData(livenessProbe.data) } : {}),
+      ...(startupProbe.enabled ? { startupProbe: constructProbeData(startupProbe.data) } : {}),
+    };
+    k8sUpdate(modelFor(referenceFor(resource.data)), updatedResource)
+      .then(() => {
+        actions.setSubmitting(false);
+        actions.setStatus({ error: '' });
+        history.push(`/topology/ns/${updatedResource.metadata.namespace}`);
+      })
+      .catch((err) => {
+        actions.setSubmitting(false);
+        actions.setStatus({ errors: err });
+      });
+  };
+
+  const initialValues = {
+    healthChecks: getHealthChecksData(resource.data),
+    containerName: container[0].name,
+    resources: getResourcesType(resource.data),
+    image: {
+      ports: container[0].ports,
+    },
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={healthChecksProbesValidationSchema}
+      onSubmit={handleSubmit}
+      onReset={history.goBack}
+    >
+      {(props) => (
+        <AddHealthCheck resource={resource} currentContainer={currentContainer} {...props} />
+      )}
+    </Formik>
+  );
+};
+
+const AddHealthCheckPage: React.FC<AddHealthCheckProps> = ({ match }) => {
+  const { ns, kind, name, containerName } = match.params;
+  const resource: FirehoseResource[] = [
+    ...(kind === ServiceModel.kind
+      ? [
+          {
+            kind: referenceForModel(ServiceModel),
+            namespace: ns,
+            isList: false,
+            name,
+            prop: 'resource',
+          },
+        ]
+      : [
+          {
+            kind,
+            namespace: ns,
+            isList: false,
+            name,
+            prop: 'resource',
+          },
+        ]),
+  ];
+  return (
+    <>
+      <Helmet>
+        <title>Health Checks</title>
+      </Helmet>
+      <div>
+        <PageHeading title="Health Checks" />
+        <Firehose resources={resource}>
+          <HealthCheckWrapper currentContainer={containerName} />
+        </Firehose>
+      </div>
+    </>
+  );
+};
+
+export default AddHealthCheckPage;
```

Comment:
These should be two different files one AddHealthCheckPage and the other HealthCheckWrapper.

Reply:
Done.